language/bytecode-verifier/src/reference_safety/mod.rs

```diff
@@ -324,6 +324,62 @@ fn execute_inner(
             let struct_def = verifier.resolver.struct_def_at(struct_inst.def)?;
             unpack(verifier, struct_def)
         }
+
+        Bytecode::VecEmpty(idx) => {
+            // since reference safety is scheduled after type safety, we know that the element type
+            // must be valid, i.e., the unwrap is safe here.
```

Comment:
@tnowacki/ @shazqadeer: do we `unwrap` or propagate an internal invariant violation in these sorts of cases elsewhere in the verifier?

Reply:
We should never panic if possible (though that isn't always the case in the verifier right now)

Should raise a `StatusCode::VERIFIER_INVARIANT_VIOLATION`